### PR TITLE
Added missing functionality to adapters

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

File: src/Reflection/Adapter/ReflectionFunction.php [hunk 1]
```diff
@@ -178,6 +178,19 @@ public function getParameters()
     /**
      * {@inheritDoc}
      */
+    public function getReturnType()
+    {
+        $returnType = $this->betterReflectionFunction->getReturnType();
+        if ($returnType === null) {
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionFunction.php]: Add an empty line before the conditiona

File: src/Reflection/Adapter/ReflectionMethod.php [hunk 2]
```diff
@@ -178,6 +178,19 @@ public function getParameters()
     /**
      * {@inheritDoc}
      */
+    public function getReturnType()
+    {
+        $returnType = $this->betterReflectionMethod->getReturnType();
+        if ($returnType === null) {
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionMethod.php]: Same, spacing

File: src/Reflection/Adapter/ReflectionFunction.php (hunk 1 above)
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionFunction.php]: Also - yodaism if possible

File: src/Reflection/Adapter/ReflectionMethod.php (hunk 2 above)
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionMethod.php]: Also - yodaism if possible

File: src/Reflection/Adapter/ReflectionParameter.php [hunk 3]
```diff
@@ -171,4 +179,26 @@ public function getDefaultValueConstantName()
...
+        return $this->betterReflectionParameter->hasType();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getType()
+    {
+        $type = $this->betterReflectionParameter->getType();
+        if ($type === null) {
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionParameter.php]: Same: spacing/yodaims

File: src/Reflection/Adapter/ReflectionFunction.php [hunk 4]
```diff
@@ -178,6 +178,19 @@ public function getParameters()
...
+            return null;
+        }
+
+        return new ReflectionType($returnType);
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionFunction.php]: Maybe `ReflectionType::fromReturnTypeOrNull(?string $returnType);`?

File: src/Reflection/Adapter/ReflectionFunction.php (hunk 1 above)
Comment (kukulich) [inline src/Reflection/Adapter/ReflectionFunction.php]: Fixed.

File: src/Reflection/Adapter/ReflectionFunction.php (hunk 4 above)
Comment (kukulich) [inline src/Reflection/Adapter/ReflectionFunction.php]: Fixed.

File: src/Reflection/Adapter/ReflectionMethod.php (hunk 2 above)
Comment (kukulich) [inline src/Reflection/Adapter/ReflectionMethod.php]: Fixed.

File: src/Reflection/Adapter/ReflectionParameter.php (hunk 3 above)
Comment (kukulich) [inline src/Reflection/Adapter/ReflectionParameter.php]: Fixed.

**Comment by Ocramius** [comment]
Thanks @kukulich!

**Pull request merged**